runtime/compiler/control/CompilationThread.cpp

```diff
@@ -8332,9 +8332,10 @@ TR::CompilationInfoPerThreadBase::wrappedCompile(J9PortLibrary *portLib, void *
                      options->setOption(TR_UseSymbolValidationManager, false);
                   }
 
-               // See if we need to inset GCR trees
-               if (!details.supportsInvalidation())
-                  {
+               // See if we need to insert GCR trees
+               if (!details.supportsInvalidation() ||
+                   options->getOptLevel() >= hot) // Workaround for a bug with GCR inserted in hot bodies
+                  {                               // Upgrades hot-->scorching should be done through sampling, not GCR
```

Comment:
I'd like to see an issue reference backing this workaround which explains why it was put in place.

Reply:
Updated the comment in the code and the commit message.